File: app/frontend/app/models/event.js
import Ember from 'ember';
import DS from 'ember-data';
import { htmlSafe } from '@ember/template';

var LearnEvent = DS.Model.extend({
  activity_id: DS.attr('string'),
  text: DS.attr('string'),
  tracked: DS.attr('date'),
  success_level: DS.attr('number'),
  notes: DS.attr('string'),
  user_id: DS.attr('string'),
  user_name: DS.attr('string'),
  face_class: function() {
    var level = this.get('success_level');
    if(level == 4) {
      return htmlSafe('face laugh');
    } else if(level == 3) {
      return htmlSafe('face happy');
    } else if(level == 2) {
      return htmlSafe('face neutral');
    } else if(level == 1) {
      return htmlSafe('face sad');
    } else {
      return null;
    }
  }.property('success_level')
});

export default LearnEvent;

File: lib/json_api/event.rb
module JsonApi::Event
  extend JsonApi::Json
  
  TYPE_KEY = 'event'
  DEFAULT_PAGE = 25
  MAX_PAGE =50
    
  def self.build_json(event, args={})
    json = {}
    json['id'] = event['unique_id']
    json['activity_id'] = event['activity_id']
    json['text'] = event['text']
    json['tracked'] = event['tracked']
    json['success_level'] = event['success_level'].to_i if event['success_level']
    json['notes'] = event['notes'] if event['notes']
    json['user_name'] = event['user_name']
    json['user_id'] = event['user_id']
    json
  end
end
